d2bs/kolbot/libs/bots/BattleOrders.js

```diff
@@ -42,20 +83,29 @@ function BattleOrders() {
 
 	Pather.moveTo(me.x + 6, me.y + 6);
 
-	var i,
+	let i,
 		tick = getTickCount(),
 		failTimer = 60;
 
 MainLoop:
 	while (true) {
 		switch (Config.BattleOrders.Mode) {
 		case 0: // Give BO
+			if (this.AmTardy()) {
+				break MainLoop;
+			}
+
 			for (i = 0; i < Config.BattleOrders.Getters.length; i += 1) {
 				while (!Misc.inMyParty(Config.BattleOrders.Getters[i]) || !getUnit(0, Config.BattleOrders.Getters[i])) {
 					if (getTickCount() - tick >= failTimer * 1000) {
 						showConsole();
 						print("ÿc1BO timeout fail.");
-						quit();
+						if (Config.BattleOrders.QuitOnFailedGive) {
```

Comment:
Why the same block code here, as in the other switch case?

Maybe store a variable saying you failed, and if so quit / break, instead of duplication code

Reply:
I'll update, good catch